bin/run-caliper.py

```diff
@@ -41,9 +58,24 @@ def main():
     update_json(networkfile, config)
 
     #run the caliper
+    config_general = load_config(CONFIG_PATH)
+    attempt = config_general['run_caliper']['attempt']
     bashfile = "run-caliper.sh"
     reportname = config.interval + "seconds-" + config.gaslimit + ".html"
-    os.system("bash " + bashfile + " " + reportname)
+    #running in number of attempts if it's failing (attempts variable getting from config)
+    for i in range(attempt):
+        #running run-caliper.sh with reportname
+        run_command = os.system("bash " + bashfile + " " + reportname)
+        if run_command == 0:
+            print("Running caliper success.")
+            break
+
+        # if it reach the maximum attempt, printing error message
+        if i == attempt-1:
+            print("Meet maximum retry. Running caliper error.")

```

Comment:
If this is the final caliper failure error I think it should directly after this message (and inside the if of course) exit(-1). If not main assumes everything ran correctly.

I just tested with all failures and even though caliper.sh returned the error, caliper.py returned success execution even after 3 retries.